src/java.base/share/classes/java/lang/invoke/CallSite.java

```diff
@@ -116,6 +176,16 @@ abstract sealed class CallSite permits ConstantCallSite, MutableCallSite, Volati
     CallSite(MethodHandle target) {
         target.type();  // null check
         this.target = target;
+        if (MethodHandleStatics.TRACE_CALLSITE) {
```

Comment:
This doesn't seem like the right place to put this. Anyone can indirectly extend CallSite, by extending e.g. MutableCallSite, but this code seems to assume that the call is going through `MethodHandleNatives.linkCallSite`? I suggest putting any tracing around that area. Both the target method handle, and the BSM seem to be available at that point in the code as well.

I also note that there's already a `TRACE_METHOD_LINKAGE` flag as well, which does some tracing of call site linking. Maybe that option could be enhanced instead of adding a new one?